File: back/src/controllers/jobController.js
import mongoose from 'mongoose';
import Job from '../models/Job.js';

export const getJobs = async (req, res) => {
  try {
    const list = await Job.find({}).sort({ order: 1 }).lean();
    res.json({ success: true, jobs: list });
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
};

export const createJob = async (req, res) => {
  try {
    const data = req.body;
    const val = data.title || data.name || 'وظيفة جديدة';
    const newJob = await Job.create({
      ...data,
      title: val,
      name: val,
      id: data.id || `job_${Date.now()}`,
    });
    res.status(201).json({ success: true, job: newJob });
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
};

export const updateJob = async (req, res) => {
  try {
    const { id } = req.params;
    const data = req.body;
    const val = data.title || data.name;
    if (val) {
      data.title = val;
      data.name = val;
    }
    const updated = await Job.findOneAndUpdate(
      getIdFilter(id),
      { $set: data },
      { new: true }
    );
    res.json({ success: true, job: updated });
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
};

export const deleteJob = async (req, res) => {
  try {
    const { id } = req.params;
    await Job.findOneAndDelete(getIdFilter(id));
    res.json({ success: true, message: 'تم حذف الوظيفة بنجاح' });
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
};

const getIdFilter = (id) => {
  if (!id) return { id: 'invalid_id_placeholder' };
  const is24Hex = mongoose.Types.ObjectId.isValid(id) && String(new mongoose.Types.ObjectId(id)) === id;
  if (is24Hex) {
    return { _id: id };
  }
  return { id };
};

File: back/Controllers/JobController.cs
using System;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using Eftekad.Backend.Data;
using Eftekad.Backend.Models;
using System.Linq;

namespace Eftekad.Backend.Controllers
{
    [ApiController]
    [Route("api/jobs")]
    public class JobController : ControllerBase
    {
        private readonly EftekadDbContext _context;

        public JobController(EftekadDbContext context)
        {
            _context = context;
        }

        [HttpGet]
        public async Task<IActionResult> GetJobs()
        {
            try
            {
                var list = await _context.Jobs.OrderBy(j => j.Order).ToListAsync();
                return Ok(new { success = true, jobs = list });
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }

        [HttpPost]
        public async Task<IActionResult> CreateJob([FromBody] Job data)
        {
            try
            {
                var val = !string.IsNullOrEmpty(data.Title) ? data.Title : (!string.IsNullOrEmpty(data.Name) ? data.Name : "وظيفة جديدة");
                data.Title = val;
                data.Name = val;
                data.Id = string.IsNullOrEmpty(data.Id) ? $"job_{DateTime.UtcNow.Ticks}" : data.Id;

                _context.Jobs.Add(data);
                await _context.SaveChangesAsync();

                return Created($"/api/jobs/{data.Id}", new { success = true, job = data });
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }

        [HttpPut("{id}")]
        public async Task<IActionResult> UpdateJob(string id, [FromBody] Job data)
        {
            try
            {
                var job = await _context.Jobs.FirstOrDefaultAsync(j => j.Id == id);
                if (job == null)
                {
                    return NotFound(new { error = "الوظيفة غير موجودة" });
                }

                var val = !string.IsNullOrEmpty(data.Title) ? data.Title : data.Name;
                if (!string.IsNullOrEmpty(val))
                {
                    job.Title = val;
                    job.Name = val;
                }

                if (data.Order != 0) job.Order = data.Order;
                if (!string.IsNullOrEmpty(data.ServiceYear)) job.ServiceYear = data.ServiceYear;

                job.UpdatedAt = DateTime.UtcNow;
                await _context.SaveChangesAsync();

                return Ok(new { success = true, job });
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }

        [HttpDelete("{id}")]
        public async Task<IActionResult> DeleteJob(string id)
        {
            try
            {
                var job = await _context.Jobs.FirstOrDefaultAsync(j => j.Id == id);
                if (job != null)
                {
                    _context.Jobs.Remove(job);
                    await _context.SaveChangesAsync();
                }
                return Ok(new { success = true, message = "تم حذف الوظيفة بنجاح" });
            }
            catch (Exception ex)
            {
                return StatusCode(500, new { error = ex.Message });
            }
        }
    }
}
